test/inductor/test_torchinductor.py

```diff
@@ -13654,6 +13654,28 @@ def forward(self, x):
         inputs = (torch.randn(4, device=self.device),)
         self.common(Model(), inputs)
 
+    @unittest.skipIf(
+        config.cpp_wrapper,
+        "cpp_wrapper samples will lead to invalid indexing",
```

Comment:
what happens when cpp_wrapper is used - does it raise a device-side assert? (and is it during compilation or during runtime?) and do you know if that is a regression or if that is a previously-existing bug?

Reply:
cpp wrapper generates examples and runs kernels using those examples. these examples have the rule - random tensor for real data, zero data for integral data (to capture the kernel cubin i think?). so if the model makes assumptions (like repeats.sum() == output_size), cpp_wrapper fails. this is why the original MR disabled the decomp for cpp_wrapper, too.